.ci/Jenkinsfile

```diff
@@ -0,0 +1,106 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+@Library('apm@current') _
+
+pipeline {
+  agent none
+  environment {
+    REPO = 'apm'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    NOTIFY_TO = credentials('notify-to')
+    JOB_GCS_BUCKET = credentials('gcs-bucket')
+    PIPELINE_LOG_LEVEL='INFO'
+  }
+  options {
+    timeout(time: 3, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  stages {
+    stage('Initializing'){
+      agent { label 'linux && immutable' }
+      options { skipDefaultCheckout() }
+      environment {
+        HOME = "${env.WORKSPACE}"
+        PATH = "${env.PATH}:${env.WORKSPACE}/bin:${env.WORKSPACE}/${env.BASE_DIR}/.ci/scripts"
+      }
+      stages {
+        stage('Checkout'){
+          agent { label 'immutable && docker' }
+          options { skipDefaultCheckout() }
+          steps {
+            deleteDir()
+            dir("${BASE_DIR}"){
+              checkout scm
```

Comment:
Is it possible to use gitCheckout step? Or is it only for MBP?

Reply:
It's possible

Reply:
When replacing that block with:

```groovy
         steps {
            deleteDir()
            gitCheckout(basedir: "${BASE_DIR}", githubNotifyFirstTimeContributor: true)
            stash allowEmpty: true, name: 'source', useDefaultExcludes: false
          }
```
I see this error in the local instance:
> No valid SCM config passed. Please double check the environment variable env.BRANCH_NAME=null is correct.

Reply:
Wait, RTFM! As I'm not in a MBP, I have to provide all information: credentialId, repo, branch, etc. Checking right now

Reply:
One thing, when defining the scm at `gitCheckout` level, then the JJBB file gets simplified, right?

Instead of:
```yaml
pipeline-scm:
      script-path: .ci/Jenkinsfile
      scm:
        - git:
            url: git@github.com:mdelapenya/apm.git
            refspec: +refs/heads/*:refs/remotes/origin/*
            wipe-workspace: 'True'
            name: origin
            shallow-clone: true
            credentials-id: f6c7695a-671e-4f4f-a331-acdce44ff9ba
            reference-repo: /var/lib/jenkins/.git-references/apm.git
            branches:
              - 'master'
```
Then  it would be just:
```yaml
pipeline-scm:
      script-path: .ci/Jenkinsfile
```

Reply:
Nevermind, the scm must be set in order to fetch the Jenkinsfile for initial clone